lib/rules/best-practises/no-unused-vars.js

```diff
@@ -138,9 +126,32 @@ class NoUnusedVarsChecker extends BaseChecker {
   }
 
   FunctionDefinition(node) {
+    if (node.body === null) {
```

Comment:
yes! everywhere a function is _declared_ but not  _defined_ (which also includes unimplemented virtual functions in abstract contracts), we choose to not apply this rule, as the parameters can't be used (since there is no body) but giving them names aids in readability and is necessary for natspec. Adding this as a comment, good catch!